zerver/tests/test_subs.py

```diff
@@ -639,13 +666,15 @@ def test_admin_remove_others_from_subbed_private_stream(self) -> None:
 
     def test_admin_remove_others_from_unsubbed_private_stream(self) -> None:
         """
-        Even if you're an admin, you can't remove people from private
+        If you're an admin, you can remove people from private
         streams you aren't on.
         """
         result = self.attempt_unsubscribe_of_principal(
-            query_count=5, is_admin=True, is_subbed=False, invite_only=True,
-            other_user_subbed=True)
-        self.assert_json_error(result, "Cannot administer invite-only streams this way")
+            query_count=21, is_admin=True, is_subbed=False, invite_only=True,
+            other_user_subbed=True, other_sub_users=[self.example_user("hamlet")])
```

Comment:
I changed "hamlet" to othello, because with your implementation, Hamlet was actually being subscribed as the "current user".

I think a good follow-up to this would be to change `attempt_unsubscribe_of_principal` to use the Iago user rather than making the hamlet user an admin if you requested `is_admin=True`; might be less confusing.